File: BoatstoSavePeople.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
       
        int total = 0;
        int n = people.size();
        int a = 0 ,b = n-1;
        sort(people.begin(),people.end());
        
        while(a <= b)
        {
            if(people[a] + people[b] <= limit)
            {
                total ++;
                a++;
                b--;
            }
            else if(people[a] > people[b])
            {
                a++;
                total++;
            }
            else
            {
                b--;
                total++;
            }
                
        }
        
        return total;
    }
};

File: BoatsToSavePeople.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:

        people.sort()

        left = 0
        right = len(people) - 1
        numOfBoats = 0

        while left <= right:
            if people[left] + people[right] <= limit:
                left += 1
                
            right -= 1
            numOfBoats += 1

        return numOfBoats
